contracts/metagov/BalancerGaugeStaker.sol

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import "../pcv/PCVDeposit.sol";
+import "./utils/LiquidityGaugeManager.sol";
+import "../external/balancer/IBalancerMinter.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/// @title Deposit that can stake in Balancer gauges
+/// @author Fei Protocol
+contract BalancerGaugeStaker is PCVDeposit, LiquidityGaugeManager {
+    using SafeERC20 for IERC20;
+
+    event BalancerMinterChanged(address indexed oldMinter, address indexed newMinter);
+
+    address private constant BAL = 0xba100000625a3754423978a60c9317c58a424e3D;
+
+    address public balancerMinter;
+
+    /// @notice Balancer gauge staker
+    /// @param _core Fei Core for reference
+    constructor(
```

Comment:
Why have a constructor that sets some state when this contract is ugpradeable and needs the `initialize()` function to be called? This constructor won't `initialize` core

Reply:
We want to support both upgradeability and non-upgradeability?